### PR TITLE
add cni benchmark: service discovery test

### PR DESCRIPTION
**Pull request by jshr-w**
This PR:
- Updates the CL2 deployment template to include a service and use RollingUpdate
- Adds a modularized and configurable load test config (with the intent that this config can be further updated for future load tests)
- Modularizes CL2 measurements and Cilium-specific measurements
- Allows user to choose CL2 config file to pass in for improved future modularity

### CODE REVIEW

File: pipelines/perf-eval/slo-servicediscovery.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    displayName: "Every 6 Hour"
+    branches:
+      include:
+        - main
+    always: true
+
+variables:
+  SCENARIO_TYPE: perf-eval
+  SCENARIO_NAME: slo-servicediscovery
+  SCENARIO_VERSION: jshr/servicechurn
```
Comment (alyssa1303) [inline pipelines/perf-eval/slo-servicediscovery.yml:13]: rename to "main"

File: steps/engine/clusterloader2/autoscale/cleanup.yml [hunk 2]
```diff
@@ -1,6 +1,11 @@
 steps:
 - script: |
-    kubectl delete --all nodeclaim --ignore-not-found
-    kubectl delete --all nodepool --ignore-not-found
+    nodeclaim_exists=$(kubectl api-resources | grep -w nodeclaim)
```
Comment (alyssa1303) [inline steps/engine/clusterloader2/autoscale/cleanup.yml:3]: we can revert this change since we use a different topology for service test now

File: steps/topology/service-churn/validate-resources.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+steps:
+  - template: /steps/cloud/${{ parameters.cloud }}/update-kubeconfig.yml
+    parameters:
+      role: slo
+      region: ${{ parameters.regions[0] }}
+  - template: /steps/engine/clusterloader2/slo/validate.yml
+    parameters:
+      desired_nodes: 1006
+  - ${{ if eq(parameters.cloud, 'aws') }}:
```
Comment (alyssa1303) [inline steps/topology/service-churn/validate-resources.yml:17]: remove this part since karpenter isn't used

File: steps/topology/service-churn/collect-clusterloader2.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  default: {}
+
+steps:
+- template: /steps/engine/clusterloader2/slo/collect.yml
+  parameters:
+    cloud: ${{ parameters.cloud }}
+    engine_input: ${{ parameters.engine_input }}
+    region: ${{ parameters.regions[0] }}
+- ${{ if eq(parameters.cloud, 'aws') }}:
+  - template: /steps/engine/clusterloader2/autoscale/cleanup.yml
```
Comment (alyssa1303) [inline steps/topology/service-churn/collect-clusterloader2.yml:19]: remove this part since karpenter isn't used

File: steps/topology/slo/validate-resources.yml [hunk 5]
```diff
@@ -15,7 +15,10 @@ steps:
     parameters:
       desired_nodes: 6
   - ${{ if eq(parameters.cloud, 'aws') }}:
-      - script: kubectl apply -f $KARPENTER_NODEPOOL_FILE
+      - script: |
```
Comment (alyssa1303) [inline steps/topology/slo/validate-resources.yml:18]: Revert this change since we use a different topology for service test now

File: scenarios/perf-eval/slo-servicediscovery/terraform-inputs/aws.tfvars [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        {
+          key    = "slo"
+          value  = "true"
+          effect = "NO_SCHEDULE"
+        }
+      ]
+    }
+  ]
+
+  eks_addons         = []
```
Comment (alyssa1303) [inline scenarios/perf-eval/slo-servicediscovery/terraform-inputs/aws.tfvars:149]: Need to enable these addon on AWS side:
```
eks_addons = [
      { name = "vpc-cni" },
      { name = "kube-proxy" },
      { name = "coredns" }
  ]
```

File: modules/python/clusterloader2/utils.py [hunk 7]
```diff
@@ -46,65 +47,70 @@ def run_cl2_command(kubeconfig, cl2_image, cl2_config_dir, cl2_report_dir, provi
 
 def get_measurement(file_path):
     file_name = os.path.basename(file_path)
+    print(file_name)
```
Comment (sumanthreddy29) [inline modules/python/clusterloader2/utils.py:50]: Can we remove this print statement?

File: steps/engine/clusterloader2/autoscale/cleanup.yml [hunk 8]
```diff
@@ -1,6 +1,6 @@
 steps:
 - script: |
-    kubectl delete --all nodeclaim --ignore-not-found
-    kubectl delete --all nodepool --ignore-not-found
+      kubectl delete --all nodeclaim --ignore-not-found
```
Comment (alyssa1303) [inline steps/engine/clusterloader2/autoscale/cleanup.yml:3]: remove extra space so the file won't show up as getting changed anymore

File: steps/topology/slo/validate-resources.yml (hunk 5 above)
Comment (alyssa1303) [reply steps/topology/slo/validate-resources.yml:18]: Can we revert to original format so the file won't show up as getting changed anymore?

**Review by alyssa1303 — APPROVED** [review]
